modules/fatfs/CMakeLists.txt

```diff
@@ -18,7 +18,10 @@ if(CONFIG_FAT_FILESYSTEM_ELM)
 
   zephyr_library_sources_ifdef(CONFIG_FS_FATFS_LFN
     ${ZEPHYR_FATFS_MODULE_DIR}/option/ffunicode.c
-    ${ZEPHYR_FATFS_MODULE_DIR}/option/ffsystem.c
+  )
+
+  zephyr_library_sources_ifdef((CONFIG_FS_FATFS_LFN OR CONFIG_FS_FATFS_REENTRANT)
+    zfs_ffsystem.c
```

Comment:
This seems to be a wrong Kconfig order, if FS_FATFS_LFN mode requires reentrancy support, then it must pull FS_FATFS_REENTRANT in, either as a depends on or select, the this line is simplified

Reply:
They are not dependent on each other. Just both helper functions are implemented in the same file (same as in the example file provided by Elm Chan)
I did not get my local test running today, so I tried to test it directly on the pipeline :see_no_evil: 

Do you have a suggestion how I can include this zfs_ffsystem.c if either CONFIG_FS_FATFS_LFN or CONFIG_FS_FATFS_REENTRANT is set?

Reply:
This is OK then, I misread how it works.